test/snap/test-ejs-with-snapshot.js

```diff
@@ -75,33 +73,32 @@ test('reply.view with ejs engine, templates with folder specified, include files
     reply.type('text/html; charset=utf-8').view('index', data)
   })
 
-  fastify.listen({ port: 0 }, err => {
-    t.error(err)
-
-    sget({
-      method: 'GET',
-      url: 'http://localhost:' + fastify.server.address().port
-    }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.equal(response.headers['content-type'], 'text/html; charset=utf-8')
-      t.equal(response.headers['content-length'], '' + body.length)
-
-      let content = null
-      ejs.renderFile(path.join(templatesFolder, 'index.ejs'), data, options, function (err, str) {
-        content = str
-        t.error(err)
-        t.equal(content.length, body.length)
-      })
-      t.matchSnapshot(content.replace(/\r?\n/g, ''), 'output') // normalize new lines for cross-platform
-
-      fastify.close()
+  await fastify.listen({ port: 0 })
+
+  const result = await fetch('http://127.0.0.1:' + fastify.server.address().port)
+
+  const responseContent = await result.text()
+
+  t.assert.strictEqual(result.status, 200)
+  t.assert.strictEqual(result.headers.get('content-length'), '' + responseContent.length)
+  t.assert.strictEqual(result.headers.get('content-type'), 'text/html; charset=utf-8')
+
+  let content = null
+  await new Promise(resolve => {
+    ejs.renderFile(path.join(templatesFolder, 'index.ejs'), data, options, function (err, str) {
+      content = str
+      t.assert.ifError(err)
+      t.assert.strictEqual(content.length, responseContent.length)
+      resolve()
     })
   })
+  t.assert.snapshot(content.replace(/\r?\n/g, '')) // normalize new lines for cross-platform
+
+  await fastify.close()
 })
 
-test('reply.view with ejs engine, templates with folder specified, include files and attributes; requires TAP snapshots enabled; page with includes', t => {
-  t.plan(8)
+test('reply.view with ejs engine, templates with folder specified, include files and attributes; requires TAP snapshots enabled; page with includes', async t => {
```

Comment:
Tap still mentioned in test descriptions.

Reply:
Since node20 doesn't support snapshotting, I've reverted back to tap for just this file